src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -781,43 +781,49 @@ protected async Task<ImmutableArray<Diagnostic>> GetSortedDiagnosticsAsync(Solut
                 var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
                 var compilationWithAnalyzers = compilation.WithAnalyzers(analyzers, GetAnalyzerOptions(project), cancellationToken);
                 var allDiagnostics = await compilationWithAnalyzers.GetAllDiagnosticsAsync().ConfigureAwait(false);
-                diagnostics.AddRange(allDiagnostics.Where(IsIncludedDiagnostic));
+
+                diagnostics.AddRange(allDiagnostics.Where(diagnostic => IsCompilerDiagnosticIncluded(diagnostic, compilerDiagnostics)));
             }
 
             var results = SortDistinctDiagnostics(diagnostics);
             return results.ToImmutableArray();
+        }
 
-            // Local function
-            bool IsIncludedDiagnostic(Diagnostic diagnostic)
-            {
-                return !IsCompilerDiagnostic(diagnostic)
-                    || IsCompilerDiagnosticIncluded(diagnostic);
-            }
+        /// <summary>
+        /// Extension point to allow inheriting classes to include diagnostics which would otherwise be filtered. The default implementation returns false
+        /// </summary>
+        /// <param name="diagnostic">the diagnostic which is subject to filter processes.</param>
+        /// <param name="compilerDiagnostics">level of diagnostic used to filter compiler diagnostics</param>
+        /// <returns>return true to exclude a diagnostic, false to leave it up to internal logic.</returns>
```

Comment:
```suggestion
        /// <returns><see langword="true"/> to include the diagnostic for validation; otherwise, <see langword="false"/> to exclude a diagnostic.</returns>
```